### DISCUSSION TITLE
Hasil coding tidak muncul-python pandas

### DISCUSSION THREAD

**Discussion by AngelaSherina** [teamalgoritma/community#299] [Q&A] [answered]
halo, saya melakukan coding untuk mencari nilai total sales tertinggi pada tahun 2017 di perusahaan mana, namun saat melakukan coding hasilnya tidak ada
![Web capture_9-9-2022_175132_localhost](https://user-images.githubusercontent.com/112871237/189337280-2666423a-da53-41cd-af78-4296ef326057.jpeg)

**Comment by taruma** [comment, marked as answer]
mungkin dari _conditional_ tersebut memang tidak ada baris yang memenuhi kriteria tersebut. 🤔

update:
- oh, `cond2` itu nilainya numerical (hasil dari `.max()`), jadi bukan dalam bentuk `pd.Series` dengan `dtype` _boolean_. Sehingga, `cond2` bisa dikoreksi menjadi `cond2 = clients['Total Sales'] == clients['Total Sales'].max()`

---

**Reply by Talioll** [reply]
Setuju sama mas taruma, untuk kondisi kedua belum ditentukan untuk penentuan kolom yg mau 'filter', langsung ke hasil yg mau dicari saja (as True).
Dari syntax kondisi 2 tertulis:
cond2 = clients['Total Sales'].max() ==> kalau diartikan (kondisi2:  nilai max Total Sales)
cond1 = clients['Year'] == 2017  ==> kalau diartikan (kondisi1: Kolom Year dengan nilai 2017)

jadi waktu waktu masuk proses return clients[cond1 & cond2]
cond1 akan mencari nilai 2017 di **kolom Year** dengan nilai **2017**,
sedang untuk cond2, belum ditunjuk kolom mana yg akan di filter (krn langsung kepada nilai yg dicari)

Karena operator dan /(&) gak bisa nemu jawabannya untuk cond 2, jadi return clients[cond1 & cond2] tidak ada yg memenuhi.

solution:
cond2 = tunjuk kolom yg akan di filter == nilai yg akan dicari
cond2 = clients['Total Sales'] == clients['Total Sales'].max()
